api/oneclick.py

```diff
@@ -0,0 +1,707 @@
+                # yes, it could be a server error, but it can also be a malformed value in the request
+                # sometimes those cause nice sql stack traces, which end up in 500s.
+                if message.startswith("eXtensible Framework encountered a SqlException"):
+                    raise InvalidInputException(action + ": " + message)
+                elif message == "A patron account with the specified username, email address, or card number already exists for this library.":
+                    raise RemotePatronCreationFailedException(action + ": " + message)
+                else:
+                    raise RemoteInitiatedServerError(message, action)
+
+            # a 409 conflict code can mean many things
+            if response.status_code == 409 and action == 'checkout':
+                if message == "Maximum checkout count reached.":
+                    raise PatronLoanLimitReached(action + ": " + message)
+                elif message == "Checkout item already exists":
+                    # we tried to borrow something the patron already has
+                    raise AlreadyCheckedOut(action + ": " + message)
+                else:
+                    raise CannotLoan(action + ": " + message)
+
+            if response.status_code == 409 and action == 'checkin':
+                if message == "Checkout does not exists or it is already terminated or expired.":
+                    # we tried to return something the patron doesn't own
+                    raise NotCheckedOut(action + ": " + message)
+                else:
+                    raise CannotReturn(action + ": " + message)
+
+            if response.status_code == 404:
+                raise NotFoundOnRemote(action + ": " + message)
+
+            if response.status_code == 400:
+                raise InvalidInputException(action + ": " + message)
+
+        elif message:
+            # http code was OK, but info wasn't sucessfully read from db
+            if message.startswith("eXtensible Framework was unable to locate the resource for RB.API.OneClick.UserPatron.Get"):
+                raise PatronNotFoundOnRemote(action + ": " + message)
+            else:
+                self.log.warning("%s not retrieved: %s ", action, message)
+                raise CirculationException(action + ": " + message)
+
+
+    def queue_response(self, status_code, headers={}, content=None):
+        """ Allows smoother faster creation of unit tests by letting 
+        us live-test as we write. """
+        pass
+
+
+
+class MockOneClickAPI(BaseMockOneClickAPI, OneClickAPI):
+    pass
+
+
+
+class OneClickCirculationMonitor(Monitor):
+    """Maintain LicensePools for OneClick titles.
+
+    Bibliographic data isn't inserted into new LicensePools until
+    we hear from the metadata wrangler.
+    """
+    def __init__(self, _db, name="OneClick Circulation Monitor",
+                 interval_seconds=500,
+                 maximum_consecutive_unchanged_books=None):
```

Comment:
I don't think you do anything with maximum_consecutive_unchanged_books; I would take it out.